examples/docker-simple-on-ubuntu-managed-disks/main.tf

```diff
@@ -0,0 +1,196 @@
+  name     = "${random_id.diagnostics_storage_account_name.keepers.resource_group}"
+  location = "${var.resource_group_location}"
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_storage_account" "diagnostics_storage_account" {
+  name                = "docker${random_id.diagnostics_storage_account_name.hex}"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  account_type = "${var.config["diagnostics_storage_account_type"]}"
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_public_ip" "public_ip" {
+  name                = "${var.config["namespace"]}-publicip"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  public_ip_address_allocation = "${var.config["network_public_ipaddress_type"]}"
+  domain_name_label            = "${var.dns_name_for_public_ip}"
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_network_security_group" "network_security_group" {
+  name                = "${var.config["namespace"]}-nsg"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  security_rule {
+    name                       = "allow_ssh"
+    description                = "Allow inbound traffic on default ssh port 22."
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "22"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_virtual_network" "virtual_network" {
+  name                = "${var.config["namespace"]}-vnet"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  address_space = ["${var.config["network_address_prefix"]}"]
+
+  subnet {
+    name           = "subnet1"
+    address_prefix = "${var.config["network_subnet_prefix"]}"
+    security_group = "${azurerm_network_security_group.network_security_group.id}"
+  }
+
+  tags {
+    Source = "Azure Quickstarts for Terraform"
+  }
+}
+
+resource "azurerm_network_interface" "network_interface" {
+  name                = "${var.config["namespace"]}-nic"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  location            = "${var.resource_group_location}"
+
+  ip_configuration {
+    name                          = "ipconfig1"
+    subnet_id                     = "${azurerm_resource_group.resource_group.id}/providers/Microsoft.Network/virtualNetworks/${var.config["namespace"]}-vnet/subnets/subnet1"
```

Comment:
why not split the subnet out and reference that by it's `id` instead? i.e.

```
resource "azurerm_virtual_network" "network" {
  ..
}

resource "azurerm_subnet" "test" {
  ...
  virtual_network_name = "${azurerm_virtual_network.network.name}"
  ...
}
```

which would allow us to reference the subnet ID via:

```
${azurerm_subnet.test.id}
```